src/main/java/com/commercetools/sync/commons/Cleanup.java

```diff
@@ -0,0 +1,226 @@
+package com.commercetools.sync.commons;
+
+import com.commercetools.sync.commons.models.FetchCustomObjectsGraphQlRequest;
+import com.commercetools.sync.commons.models.ResourceKeyId;
+import com.commercetools.sync.services.impl.UnresolvedReferencesServiceImpl;
+import com.commercetools.sync.services.impl.UnresolvedTransitionsServiceImpl;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.client.NotFoundException;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.commands.CustomObjectDeleteCommand;
+
+import javax.annotation.Nonnull;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import static com.commercetools.sync.commons.utils.CtpQueryUtils.queryAll;
+import static java.lang.String.format;
+
+public class Cleanup {
```

Comment:
Cleanup looks to me very generic name. Can we have some other name which explains, What exactly this class cleans up?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/651/commits/e4844012673390d8b74d3ad108f39ba50f6ae279